sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -848,11 +857,17 @@ protected Symbol visitSubqueryExpression(SubqueryExpression node, ExpressionAnal
              */
             Field field = fields.get(0);
             SingleColumnTableType singleColumnTableType = new SingleColumnTableType(field.valueType());
-            SelectSymbol selectSymbol = new SelectSymbol(relation, singleColumnTableType);
+            Symbol symbolToReturn = new SelectSymbol(relation, singleColumnTableType);
+            if (context.isSubqueryArrayExpression(node)) {
+                return symbolToReturn;
+            }
+            // A SubQuery can return more than one row. We don't allow multiple rows,
+            // except in ANY or IN expressions. Thus, we wrap the result into a function
+            // which extracts the first element and checks if there are more than one element.
```

Comment:
see comment above: creating an array is up to the array functions, this block could always return a selectsymbol.